chart/templates/ingress.yaml

```diff
@@ -102,7 +102,9 @@ spec:
     - http01:
         ingress:
           ingressClassName: {{ .Values.ingress_class | default "nginx" }}
+          {{- if .Values.ingress.includeClassAnno }}
```

Comment:
```suggestion
          {{- if .Values.ingress. useOldClassField }}
```

Reply:
Hm, seems like this should be an if / else, using ingressClassName only if useOldClassField is not set?

Reply:
Why?
```mermaid
flowchart TD
    A[Start] --> B
    B{useOldClassField is set}
    B -->|No| C['class:' will not be rendered]
    B -->|Yes| D
    D{useOldClassField is true}
    D -->|No| C['class:' will not be rendered]
    D -->|Yes| F
    F{ingress_class is set}
    F -->|No| G[ingress_class: nginx,<br>class: nginx]
    F -->|Yes, some string| H[ingress_class: some string,<br>class: some string]
```

Reply:
Oh, I see now.
Right now everyone have a config without `useOldClassField` and after this update they will have their `class:` option removed.
Right?

So. Should we name this option `useNewClassField` instead?

Reply:
Yes, I think the old `class` value is for the annotation, which is now obsolete, so can default it to off.
I'm surprised that this worked before, as this check seems to be here for a while:
https://github.com/cert-manager/cert-manager/blame/b96e0af16d197f021904b11066d6307504524f6f/pkg/issuer/acme/http/ingress.go#L159

Elsewhere, we already use `ingressClassName` so I think that should be the default, and only use the old `class` if the `useOldClassField` is set.

Maybe `useOldClassAnnotation` is more clear even. It was deprecated a while ago now (1.18): https://kubernetes.io/docs/concepts/services-networking/ingress/#deprecated-annotation

Perhaps the only reason to keep it is for GKE, which still requires it: https://cloud.google.com/kubernetes-engine/docs/concepts/ingress

Reply:
Probably to be consistent, we should update the other two ingressClassName to also only be set if `useOldClassAnnotation` is false, which will be default, otherwise set `kubernetes.io/ingress.class`
here: https://github.com/webrecorder/browsertrix/blob/main/chart/templates/ingress.yaml#L24
and https://github.com/webrecorder/browsertrix/blob/main/chart/templates/ingress.yaml#L65